esp/bindings/http/platform/httpservice.cpp

```diff
@@ -377,18 +377,24 @@ int CEspHttpServer::processRequest()
 
             checkSetCORSAllowOrigin(thebinding, m_request, m_response);
 
-            if (thebinding!=NULL)
+            if (thebinding && thebinding->isUnrestrictedSSType(stype))
             {
-                if (thebinding->isUnrestrictedSSType(stype))
-                {
-                    thebinding->onGetUnrestricted(m_request.get(), m_response.get(), serviceName.str(), methodName.str(), stype);
-                    ctx->addTraceSummaryTimeStamp(LogMin, "handleHttp");
-                    return 0;
-                }
+                //Avoid creating a span for unrestrictedSSType requests
+                thebinding->onGetUnrestricted(m_request.get(), m_response.get(), serviceName.str(), methodName.str(), stype);
+                ctx->addTraceSummaryTimeStamp(LogMin, "handleHttp");
+                return 0;
+            }
 
-                //Avoids unrestrictedSSType requests
-                m_request->startSpan();
+            //The context will be destroyed when this request is destroyed. So rather than using
+            //EspContextSpanScope spanContext(*ctx, serverSpan);
+            //which would remove the activeSpan when this function exits, use
+            //setActiveSpan()
+            //It is possible that using EspContextSpanScope would be better
+            Owned<ISpan> serverSpan = m_request->createServerSpan();
```

Comment:
I changed this to create the span, but not set it in the context because there was no clean way to restore the span, and the HttpRequest class is used in other contexts than esp.  For the moment I have left it using setActiveSpan, but it can now be changed to restore at the end of the function.
The correct behaviour may well be tied in with the behaviour of txsummary.